Bonus guide: Download the Bitcoin white paper directly from your own blockchain

VajraOfIndra opened the pull request:
#### What

A PR for a new bonus guide in the '+ Bitcoin' bonus section. The guide explains how to:

1) Download and recreate the bitcoin white paper from the user own's blockchain data since it has been encoded in a bitcoin transaction back in 2013
2) Send the recreated PDF to the user local computer to be read
3) Create a Telegram bot and send the PDF to the bot
4) Create a bash script that can be invoked at any time to send the paper to the TG bot 

### Why

1) *Demonstrate the importance of bitcoin's decentralization*: This guide shows the user the power of decentralization as no matter how censored the white paper could become (as we already saw in the news), anyone running their own full node can recreate the white paper and share it.

2) *Instructive*: Teaches the user how to create 1) a shell script, 2) a Telegram bot

3) *Fun*: Raising awareness of a nice little historical fact of the bitcoin story (i.e. the encoding of the white paper in the blockchain back in 2013)

#### How

I used this Bitcoin StackExchange thread as the main inspiration for the guide: [https://bitcoin.stackexchange.com/questions/35959/how-is-the-whitepaper-decoded-from-the-blockchain-tx-with-1000x-m-of-n-multisi/35970#35970](https://bitcoin.stackexchange.com/questions/35959/how-is-the-whitepaper-decoded-from-the-blockchain-tx-with-1000x-m-of-n-multisi/35970#35970)

#### Scope

- [ ] significant change to core configuration
- [x] independent bonus guide
- [ ] simple bug fix

#### Test & maintenance

Test: Go through the guide and try all commands

Maintenance: Minimal to none

#### Animated GIF (optional)

![white-paper](https://media.giphy.com/media/CRU9XWykMZr0th8z3C/giphy.gif)

VajraOfIndra commented:
I'm sure the wording could be better in some places and there are maybe additional background story information I missed and should be included.

Anyways, any suggestions to improve this guide are very welcomed.

Stadicus commented on bonus/bitcoin/white-paper.md:
```diff
@@ -0,0 +1,137 @@
+Status: Tested v3
+{: .label .label-green }
+
+![white paper transaction](../../images/white-paper-transac.png)
```
The guide is mainly about getting the Bitcoin whitepaper on the command line. I think showing the BTC RPC Explorer as the main image is a bit off target.

VajraOfIndra replied:
True.. I opted for this pic because it was a bit nicer looking than a CLI screenshot. 
What nice picture could we use instead? maybe just the top part of the white paper with a CLI command above? any other/better looking idea?

Stadicus replied:
Maybe just the whitepaper itself? 

![bitcoin-whitepaper](https://user-images.githubusercontent.com/19550140/145648926-998a3f95-0902-4aa4-8f2f-9197f9efcf92.png)

> PS: we could move the Explorer pic to the Explorer section

Good idea

VajraOfIndra replied:
Neat! could you send me this pic with the pink border via TG?

VajraOfIndra replied:
PS: I used the pic just above, the resolution seems allright.

Stadicus commented on bonus/bitcoin/white-paper.md:
```diff
@@ -0,0 +1,137 @@
+### Trivia: Read the paper in the BTC-RPC-Explorer 
+
+The BTC-RPC-Explorer has also a functionality to extract the data from the node and display the PDF in the web browser.
+* Open your [BTC RPC Explorer](https://raspibolt.org/btcrpcexplorer.html)
```
Shouldn't this link point to `https://raspibolt.local:4000/`?

VajraOfIndra replied:
I think I chose the link to the guide because some users might not have installed the Explorer yet but it's fairly safe to assume that most would have it installed so your suggestion makes more sense.  I changed the link accordingly.

Stadicus replied:
Ok. We could also link to the guide, but then I'd change the name of the link a bit to make this clear.

VajraOfIndra replied:
I think that's enough, users can easily find the install guide if they didn't install it yet.

Stadicus reviewed (approved):
Lovely guide, no blockers from my end. I was so free to add a lot of little (optional) nits anyway. :)

See all the fixes in my dev branch:  
https://github.com/Stadicus/raspibolt-dev/commits/pr/827

Overall, I feel the ToC structure could be a bit clearer, as the guide shows two main ways to get the whitepaper: hard on the command line, and easy using the BTC RPC Explorer

* Introduction
* Extract the whitepaper directly from Bitcoin Core
  * Send the PDF to your desktop computer
  * Send the PDF to your Telegram account
* Get the whitepaper from BTC RPC Explorer

In that context, I'm not sure if the bash script adds much value, as I guess sending the whitepaper is a one-time thing.

I also added a section on how the long extraction command works, and instructions on how to copy the pdf with Windows.

Feel free to pick what you want, I can also open a separate pull request after to add these additional sections.

VajraOfIndra commented on bonus/bitcoin/white-paper.md:
```diff
@@ -0,0 +1,137 @@
+
+The transaction contains 947 outputs and the sender spent almost 60 millions sats as miner fee! Some explanations on how the paper is encoded in the transaction is given in a [Bitcoin stackexchange post](https://bitcoin.stackexchange.com/questions/35959/how-is-the-whitepaper-decoded-from-the-blockchain-tx-with-1000x-m-of-n-multisi/35970#35970){:target="_blank"} from 2015.  
+
+This guide explains how to reconstruct the Bitcoin white paper PDF using data from your own verified blockchain. No matter how censored the white paper could become (e.g. see [here](https://bitcoinmagazine.com/business/copa-suing-craig-wright-over-bitcoin-white-paper-claims){:target="_blank"}), you will know how to recreate and share the foundational document of Bitcoin.
```
Good point, I'll keep this in mind for future guides!

VajraOfIndra commented:
> Lovely guide, no blockers from my end. I was so free to add a lot of little (optional) nits anyway. :)

Great stuff 👍 

> See all the fixes in my dev branch: https://github.com/Stadicus/raspibolt-dev/commits/pr/827

Awesome, all very good! I went through all of them and implemented them all. PTAL!

> Overall, I feel the ToC structure could be a bit clearer, as the guide shows two main ways to get the whitepaper: hard on the command line, and easy using the BTC RPC Explorer
> 
>     * Introduction
> 
>     * Extract the whitepaper directly from Bitcoin Core
>       
>       * Send the PDF to your desktop computer
>       * Send the PDF to your Telegram account
> 
>     * Get the whitepaper from BTC RPC Explorer

I like this structure!
 
> In that context, I'm not sure if the bash script adds much value, as I guess sending the whitepaper is a one-time thing.

I thought it might be a good occasion to teach users how to create a simple bash script that combines two commands into one. I removed it.
But I quite like the idea of having simple tutorials that show how to use useful Linux tools spread out in the various bonus guides (e.g. how to create a cron job in the charge-lnd guide, how ot create a bash script somewhere else etc etc).
 
> I also added a section on how the long extraction command works, and instructions on how to copy the pdf with Windows.

That's excellent!! was very instructive for me to read this as I didn't take the time yet to parse the command in details. That's a great addition, thank you very much 💯
The Windows insructions is great addition as well.

> Feel free to pick what you want, I can also open a separate pull request after to add these additional sections.

All updated now, PTAL :) The only thing left to change is the picture at the top, see comment below)

Stadicus commented:
> I thought it might be a good occasion to teach users how to create a simple bash script

I agree, and it makes sense from this point of view. I don't mind at all if you leave it in.

Stadicus commented on bonus/bitcoin/white-paper.md:
```diff
@@ -0,0 +1,152 @@
+
+* **macOS or Linux**
+
+On your local computer, open a terminal window and type the following command. Replace YourNodeIP with the Raspberry Pi IP address (or `raspibolt.local` if it works) and do not forget the dot at the end of the line (representing the destination of the file, here the 'Home' folder of your local computer).
```
This line should be indented with 2 spaces, otherwise it's not aligned with the bullet point title

VajraOfIndra replied:
Aaaah ok, I thought it was a typo! I re-added  the indentations now.

Stadicus commented on bonus/bitcoin/white-paper.md:
```diff
@@ -0,0 +1,152 @@
+
+* **Windows**
+
+This also works with the PSCP tool from PuTTY that you can run from the command line.
```
Same here, should be indented

VajraOfIndra replied:
Done! see above

Stadicus reviewed (approved):
Nice! Two little nits regarding correct indentation, otherwise LGTM!

VajraOfIndra commented:
> Nice! Two little nits regarding correct indentation, otherwise LGTM!

Great, I'll wait to get the pic from you and I'll merge the PR once I updated the pics.
